controller/src/main/java/org/jboss/as/controller/SimpleResourceDefinition.java

```diff
@@ -225,6 +261,29 @@ public SimpleResourceDefinition(final PathElement pathElement, final ResourceDes
                                     final OperationStepHandler addHandler, final OperationStepHandler removeHandler,
                                     final OperationEntry.Flag addRestartLevel, final OperationEntry.Flag removeRestartLevel,
                                     final DeprecationData deprecationData, final boolean runtime) {
+        this(pathElement, descriptionResolver, addHandler, removeHandler, addRestartLevel, removeRestartLevel,
+                deprecationData, runtime, false);
+    }
+
+    /**
+     * {@link ResourceDefinition} that uses the given {code descriptionResolver} to configure a
+     * {@link DefaultResourceDescriptionProvider} to describe the resource.
+     *
+     * @param pathElement         the path. Can be {@code null}.
+     * @param descriptionResolver the description resolver to use in the description provider. Cannot be {@code null}      *
+     * @param addHandler          a handler to {@link #registerOperations(ManagementResourceRegistration) register} for the resource "add" operation.
+     *                            Can be {null}
+     * @param removeHandler       a handler to {@link #registerOperations(ManagementResourceRegistration) register} for the resource "remove" operation.
+     *                            Can be {null}
+     * @param deprecationData     Information describing deprecation of this resource. Can be {@code null} if the resource isn't deprecated.
+     * @param runtime             Whether this is a runtime resource
+     * @param orderedChild  Whether this child type is ordered within the parent
+     * @throws IllegalArgumentException if {@code descriptionResolver} is {@code null}.
+     */
+    public SimpleResourceDefinition(final PathElement pathElement, final ResourceDescriptionResolver descriptionResolver,
+                                    final OperationStepHandler addHandler, final OperationStepHandler removeHandler,
+                                    final OperationEntry.Flag addRestartLevel, final OperationEntry.Flag removeRestartLevel,
+                                    final DeprecationData deprecationData, final boolean runtime, final boolean orderedChild) {
```

Comment:
It would make sense but in most cases we extend the SimpleResourceDefinition and pass params in super constructor.